css/datagrid/datagrid.css

```diff
@@ -139,11 +257,17 @@
 
 .highcharts-datagrid-table thead th {
     position: relative;
-    border-bottom: 1px solid var(--highcharts-neutral-color-20);
+    border-right: var(--hcdg-header-column-border);
+    border-bottom: var(--hcdg-header-row-border);
+    background-color: var(--hcdg-header-row-background-color);
+    padding: var(--hcdg-header-padding);
 }
 
 .highcharts-datagrid-table thead th .highcharts-datagrid-header-cell-content {
```

Comment:
The reason for this is... the sorting icon on the right. The right padding is applied to avoid overlapping there, and the left padding to keep the text in the center.

Reply:
Hm, we need to fix this somehow, and only in very rare edge-cases should users need to take care of things themselves. I can have a look, if you want.